CRM/Core/Payment/AuthorizeNet.php

```diff
@@ -623,45 +624,39 @@ public function cancelSubscription(&$message = '', $params = []) {
 
     $template->assign('apiLogin', $this->_getParam('apiLogin'));
     $template->assign('paymentKey', $this->_getParam('paymentKey'));
-    $template->assign('subscriptionId', CRM_Utils_Array::value('subscriptionId', $params));
+    $template->assign('subscriptionId', $params['subscriptionId']);
```

Comment:
I did this change for 2 reasons
1) it should throw an enotice if not present as it is required but more importantly
2) to demonstrate in tested code that even though this function is not aware that it is receiving a PropertyBag & has made no changes to support it - it still retrieves the value as it always has